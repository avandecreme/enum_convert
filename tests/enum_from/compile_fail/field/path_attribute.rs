use enum_convert::EnumFrom;

enum Source {
    Struct {
        x: i32
    }
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from]
    Struct {
        #[enum_from] // Should be #[enum_from(Source::Struct.x)]
        a: i32
    },
}

fn main() {}
