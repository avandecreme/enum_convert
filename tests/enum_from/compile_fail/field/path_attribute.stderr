error: expected a list of field names, for example #[enum_from(Source1::VariantA.field_x, Source2::VariantB.field_y)]
  --> tests/enum_from/compile_fail/field/path_attribute.rs:12:9
   |
12 |         #[enum_from] // Should be #[enum_from(Source::Struct.x)]
   |         ^
